### PR TITLE
Azure: Separate VMSS rate limiting

### PR DESCRIPTION
**Pull request by sylr**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

We need it to fix errors we have when the global rate limiting allows kubernetes to query VMSS API beyond the low limits it enforces.

**Which issue(s) this PR fixes**:

https://github.com/Azure/AKS/issues/1187

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Azure: Separate VMSS rate limiting
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by sylr** [comment]
cc @andyzhangx

File: pkg/cloudprovider/providers/azure/azure.go [hunk 1]
```diff
@@ -51,10 +51,13 @@ const (
 	CloudProviderName      = "azure"
 	rateLimitQPSDefault    = 1.0
 	rateLimitBucketDefault = 5
-	backoffRetriesDefault  = 6
-	backoffExponentDefault = 1.5
-	backoffDurationDefault = 5 // in seconds
-	backoffJitterDefault   = 1.0
+	// VMSS API limits are 180 per 3 minutes & 900 per 30 minutes
```
Comment (andyzhangx) [inline pkg/cloudprovider/providers/azure/azure.go]: this PR would add the cloud provider rate limit, and `180 per 3 minutes & 900 per 30 minutes` are for the whole subscription. It's hard to tell the correct rate limit value per vmss cluster.

**Comment by andyzhangx** [comment]
@sylr this is a cherry-pick, usually we would do the fix in master and then cherry-pick to old release, that's the standard procedure.

/hold

**Comment by sylr** [comment]
> @sylr this is a cherry-pick, usually we would do the fix in master and then cherry-pick to old release, that's the standard procedure.
> 
> /hold

I made another PR for master https://github.com/kubernetes/kubernetes/pull/85949

**Comment by feiskyer** [comment]
let's work on master first.
/close

**Pull request closed without merging**